### PR TITLE
[Feature][Doc] Improved connectors v2 contribution guide

### LINKED ISSUE #2541: [Feature][Doc] The contribute document need be update
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The contribute document in `seatunnel/docs/en/contribution/contribute-plugin.md` are out of date. We should add connector-v2 contribute guide in it. And fix `plugin-mapping.properties` path in doc. And add any information which can help new developer start.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Mr-LiuXu**
..was encountered
https://github.com/apache/incubator-seatunnel/issues/2541

## Purpose of this pull request

  Improve the connector-v2 contribution guide and add common basic modules and formats to help new developers enter development more quickly
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Mr-LiuXu** [comment]
I have improved some details and debugging steps on the contribution guide of seatunnel-connectors-v2 and added some common basic modules, formats, etc. Can you take a look?

File: seatunnel-connectors-v2/README.md [hunk 1]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
 In order to separate from the old code, we have defined new modules for execution flow. This facilitates parallel
 development at the current stage, and reduces the difficulty of merging.
 
+### engineering structure
+
+-  ../`seatunnel-connectors-v2`                                        connector code implementation
+-  ../`seatunnel-translation`                                          translation layer for the new connector
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../`seatunnel-translation`                                          translation layer for the connector-v2
```

File: seatunnel-connectors-v2/README.md [hunk 2]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
 In order to separate from the old code, we have defined new modules for execution flow. This facilitates parallel
 development at the current stage, and reduces the difficulty of merging.
 
+### engineering structure
+
+-  ../`seatunnel-connectors-v2`                                        connector code implementation
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../`seatunnel-connectors-v2`                                        connector-v2 code implementation
```

File: seatunnel-connectors-v2/README.md [hunk 3]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
...
+-  ../`seatunnel-translation`                                          translation layer for the new connector 
+-  ../seatunnel-e2e/`seatunnel-flink-connector-v2-e2e`                 testcase running on flink
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../seatunnel-e2e/`seatunnel-flink-connector-v2-e2e`                 end to end testcase running on flink
```

File: seatunnel-connectors-v2/README.md [hunk 4]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
...
+-  ../seatunnel-e2e/`seatunnel-spark-connector-v2-e2e`                 testcase running on spark
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../seatunnel-e2e/`seatunnel-spark-connector-v2-e2e`                 end to end testcase running on spark
```

File: seatunnel-connectors-v2/README.md [hunk 5]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
...
+-  ../seatunnel-examples/`seatunnel-flink-connector-v2-example`        flink local running instance
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../seatunnel-examples/`seatunnel-flink-connector-v2-example`        seatunnel connector-v2 example use flink local running instance
```

File: seatunnel-connectors-v2/README.md [hunk 6]
```diff
@@ -10,6 +10,15 @@ this [issue](https://github.com/apache/incubator-seatunnel/issues/1608) for deta
...
+-  ../seatunnel-examples/`seatunnel-spark-connector-v2-example`        spark local running instance
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
-  ../seatunnel-examples/`seatunnel-spark-connector-v2-example`        seatunnel connector-v2 example use spark local running instance
```

File: seatunnel-connectors-v2/README.md [hunk 7]
```diff
@@ -22,13 +31,27 @@ configuration files used in example are saved in the "resources/examples" folder
...
    dependency.
 2. Find the dependency in your connector pom file which scope is test or provided and then add them to
    seatunnel-examples/seatunnel-flink-connector-v2-example/pom.xml(or add it to
    seatunnel-examples/seatunnel-spark-connector-v2-example/pom.xml) file and modify the scope to compile.
-3. Refer to the SeaTunnelApiExample class to develop your sample code.
+3. Add the task configuration file under resources/examples.
+4. Configure the file in the `SeaTunnelApiExample` main method.
+5. Just run the main method.
+
+### **Create the format of the connector**
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
### **Create new seatunnel v2 connector**
```

File: seatunnel-connectors-v2/README.md [hunk 8]
```diff
@@ -22,13 +31,27 @@ configuration files used in example are saved in the "resources/examples" folder
...
+
+1.Create a new module under the `seatunnel-connectors-v2` directory and name it connector - {connector name}.
+
+2.The pom file can refer to the pom file of the existing connector, and add the current sub model to the pom file of the parent model
+
+3.Create two packages corresponding to source and sink
+
+​    package org.apache.seatunnel.connectors.seatunnel.{connector name}}.source
+
+​    package org.apache.seatunnel.connectors.seatunnel.{connector name}}.sink
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md:54]: Not only this, also should add doc about:
1. add connector to plugin-mapping file
2. add connector to `seatunnel-dist/pom.xml`

Comment (Mr-LiuXu) [reply seatunnel-connectors-v2/README.md:54]: OK, I'll revise it later

**Comment by Mr-LiuXu** [comment]
I have improved the document, can you take a look.

File: seatunnel-connectors-v2/README.md [hunk 9]
```diff
@@ -22,13 +31,31 @@ configuration files used in example are saved in the "resources/examples" folder
...
+
+4.add connector to plugin-mapping file.
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
4.add connector info to plugin-mapping.properties file in seatunnel root path.
```

File: seatunnel-connectors-v2/README.md [hunk 10]
```diff
@@ -22,13 +31,31 @@ configuration files used in example are saved in the "resources/examples" folder
...
+
+5.add connector to seatunnel-dist/pom.xml.
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.md]:
```suggestion
5.add connector dependency to seatunnel-dist/pom.xml, so the connector jar can be find in binary package .
```

**Comment by Mr-LiuXu** [comment]
I have improved the document, can you take a look.

File: seatunnel-connectors-v2/README.zh.md [hunk 11]
```diff
@@ -18,7 +27,25 @@ SeaTunnel为与计算引擎进行解耦，设计了新的连接器API，通过
...
+
+2.pom文件可以参考已有连接器的pom文件，并在父model的pom文件中添加当前子model.
+
+3.新建两个package分别对应source和sink
+
+​       package org.apache.seatunnel.connectors.seatunnel.{连接器名}.source
+
+​       package org.apache.seatunnel.connectors.seatunnel.{连接器名}.sink
+
+4.将连接器添加到plugin-mapping文件.
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/README.zh.md]:
```suggestion
4.将连接器信息添加到在项目根目录的plugin-mapping.properties文件中.
```

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**